### PR TITLE
Rustup to rust-lang/rust#67707

### PR DESCRIPTION
**Pull request by JohnTitor**
`syntax_pos` -> `rustc_span`

changelog: none

### CODE REVIEW

File: doc/adding_lints.md [hunk 1]
```diff
@@ -432,12 +432,12 @@ don't hesitate to ask on Discord, IRC or in the issue/PR.
 [diagnostics]: https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/utils/diagnostics.rs
 [utils]: https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/utils/mod.rs
 [ident]: https://doc.rust-lang.org/nightly/nightly-rustc/syntax/source_map/symbol/struct.Ident.html
-[span]: https://doc.rust-lang.org/nightly/nightly-rustc/syntax_pos/struct.Span.html
+[span]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/struct.Span.html
```
Comment (JohnTitor) [inline doc/adding_lints.md]: The link points to wrong place at now, but it should work well once new nightly is landed.

Comment (JohnTitor) [reply doc/adding_lints.md]: It returns correct place now.

**Comment by JohnTitor** [comment]
It seems a tiny change from the latest rustc, fixed.

**Comment by phansch** [comment]
thanks again!

@bors r+

**Comment by JohnTitor** [comment]
Hm, it's spurious?

**Comment by phansch** [comment]
Not really sure: `thread 'integration_test' panicked at 'error: E0463', tests/integration.rs:61:9`. Let's retry.

@bors retry

**Comment by phansch** [comment]
Ah, RLS will probably also need a similar Rustup. It's fine to ignore that integration test for now.

**Comment by JohnTitor** [comment]
Got it, let's ignore then.

**Pull request merged**